### ISSUE TITLE
Some components do not set ID on wrapper element

### ISSUE THREAD

**Issue by jonotrujillo** [klarna/ui#209]
The behaviour of components when given an ID prop is not consistent. I believe that the wrapper element of the rendered component is the on that should have this attribute.

At this point I have not identified all components with broken behaviour.

### Working Example: Field

```html
<Field id='el-field' label='Enter your email' />
```

renders...

```html
<input class="field__input-1cgXQ" value="" id="el-field">
```

### Broken Example: Checklist

```html
<Checklist.Main id='el-checklist'>
  <Checklist.Item>
    Just one click and you're done
  </Checklist.Item>
  <Checklist.Item>
    Very little hassle
  </Checklist.Item>
  <Checklist.Item>
    Just do it! It can be done today, so why wait for  tomorrow?
  </Checklist.Item>
</Checklist.Main>

```

renders...

```html
<ul class="checklist-mgkV5">
    <li class="checklist__item-38jDN"><svg class="checklist__checkmark-1JR2A" viewBox="0 0 25 25" aria-labelledby="Checkmark" height="20px" width="20px"><path d="M5 13.69l4.49 4.23L19.37 8"></path></svg>
        <!-- react-text: 146 -->Just one click and you're done
        <!-- /react-text -->
    </li>
    <li class="checklist__item-38jDN"><svg class="checklist__checkmark-1JR2A" viewBox="0 0 25 25" aria-labelledby="Checkmark" height="20px" width="20px"><path d="M5 13.69l4.49 4.23L19.37 8"></path></svg>
        <!-- react-text: 150 -->Very little hassle
        <!-- /react-text -->
    </li>
    <li class="checklist__item-38jDN"><svg class="checklist__checkmark-1JR2A" viewBox="0 0 25 25" aria-labelledby="Checkmark" height="20px" width="20px"><path d="M5 13.69l4.49 4.23L19.37 8"></path></svg>
        <!-- react-text: 154 -->Just do it! It can be done today, so why wait for tomorrow?
        <!-- /react-text -->
    </li>
</ul>
```

**Comment by xaviervia** [comment]
👍 

I want to started on this, but it struck me that I’m not sure of how to represent and test this feature in the Showroom. Any ideas?

**Comment by jonotrujillo** [comment]
Since it's not really a feature, but a convention... Should there be a conventions page in the showroom?

**Comment by xaviervia** [comment]
Oj... yes there probably should. Hm

**Comment by Nevon** [comment]
Fixed by #210
